### DISCUSSION TITLE
Prioritizing the SSH connection over modprobe

### DISCUSSION THREAD

**Discussion by stauffenberg2020** [marcone/teslausb#841] [General]
Hi @marcone ,

When the ssh port is in [use](https://www.cyberciti.biz/faq/how-do-i-check-if-a-port-is-in-use-on-linux/), can we somehow delay (in a timed loop) presenting (modprobe modprobe g_mass_storage) the USB OTG

**Comment by marcone** [comment]
The thing you linked to for checking whether a port is "in use" just shows whether something is listening on a port. Since ssh is enabled, that will always be true, even if nobody is logged in via ssh, which I assume is what you actually want to check.
What are you actually trying to achieve though?

Also note that TeslaUSB doesn't use g_mass_storage.

**Comment by stauffenberg2020** [comment]
> The thing you linked to for checking whether a port is "in use" just shows whether something is listening on a port. Since ssh is enabled, that will always be true, even if nobody is logged in via ssh, which I assume is what you actually want to check. 

Yes, I want to check if anyone has logged in via ssh

> What are you actually trying to achieve though?

and if any user has connected/logged through ssh, Then I do not want teslausb to present the OTG drive to the car, Instead I want all those video/audio files to be accessible by the linux until the user has logged out of ssh, Thereby allowing other configuration changes, operations of the videos by linux

**Reply by marcone** [reply]
Why not just disable drive emulation only when you actually need to disable it, instead of disabling it the entire time somebody is logged in?
I don't know what you're actually trying to achieve, but it seems to me like you're going about it the wrong way.

**Reply by stauffenberg2020** [reply]
>Why not just disable drive emulation only when you actually need to disable it

Can you please let me know how to do this?

**Reply by marcone** [reply]
`/root/bin/disable_gadget.sh`

**Reply by stauffenberg2020** [reply]
Okay, Rather than disabling the gadget, What is the way/option to ensure that linux terminal is always accessible? I do not want the linux terminal to be disconnected often after the OTG gadget is presented to the car

**Reply by marcone** [reply]
I don't understand what you mean. Your ssh connection and the Linux terminal/shell have nothing to do with USB gadget emulation.
